server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -109,7 +109,8 @@ private void messageReceived(TcpChannel channel, InboundMessage message, long st
 
         TransportResponseHandler<?> responseHandler = null;
         ThreadContext threadContext = threadPool.getThreadContext();
-        try (ThreadContext.StoredContext existing = threadContext.stashContext()) {
+        assert threadContext.isDefaultContext();
+        try (var ignored = threadContext.newStoredContext()) {
```

Comment:
Just checking: why is this changed?
I assume we don't care about preserving headers here, as we set them explicitly just at the next step?

Reply:
Yes right, no need to muck around creating the stashed context when we're overwriting its contents straight away. But anyway it's a bug not to be in default context here so there's no difference.